control-plane/agents/src/bin/core/volume/specs.rs

```diff
@@ -327,6 +327,27 @@ impl ResourceSpecs {
         self.volumes.values().map(|v| v.lock().clone()).collect()
     }
 
+    /// Gets all VolumeSnapshot Specs.
+    pub(crate) fn snapshots(&self) -> Vec<VolumeSnapshot> {
+        self.volume_snapshots.values().map(|v| v.lock().clone()).collect()
+    }
+
+    /// Gets all VolumeSnapshot Specs, filtered by volume id.
+    pub(crate) fn snapshots_by_vol(&self, filter_by: SpecFilterParam<VolumeId>) -> Vec<VolumeSnapshot> {
+        self.volume_snapshots.values().filter_map(|v| if v.immutable_ref().spec().source_id() == filter_by.filter() {
+            Some(v.lock().clone())
+        } else { None })
+        .collect()
+    }
+
+    /// Gets all VolumeSnapshot Specs, filtered by snapshot id.
+    pub(crate) fn snapshot_by_snapid(&self, filter_by: SpecFilterParam<SnapshotId>) -> Vec<VolumeSnapshot> {
+        self.volume_snapshots.values().filter_map(|v| if v.immutable_ref().spec().uuid() == filter_by.filter() {
+            Some(v.lock().clone())
+        } else { None })
+        .collect()
```

Comment:
Why do we need a filter by snapshot id? Can we not simply get from the ResourceMap with snapshot id as key? Also why would be get a vector here if we are getting by snapshot id?